docs/UserGuide.md

```diff
@@ -1,42 +1,212 @@
+
+**Notes about the command format:**<br>
+
+* Words in `UPPER_CASE` are the parameters to be supplied by the user.<br>
+  e.g. in `deposit AMOUNT`, `AMOUNT` is a parameter which can be used as `deposit 300`.
+
+* Parameters MUST be in the specified order for the command to take appropriate effect.<br>
+
+
+* Extraneous parameters for commands that do not take in parameters (such as `help`, `view-transactions-all`, `exit`) will be ignored.<br>
+  e.g. if the command specifies `help 123`, it will be interpreted as `help`.
+
+</div>
+
+### Viewing help : `help`
+
+Shows a list of all the commands available and what they attempt to do.
+
+Format: `help`
+
+
+### Adding an account: `add-account`
+
+Initiates Adding a new account. When calling the command there is no requirement for any parameter.
+However, once the command starts, it will lead you through the process of a new account creation
+prompting your name and balance to be stored.
+
+Format: `add-account` <br />
+CLI prompts: "Whats your name?" <br />
+your input: `NAME` <br />
+CLI prompts: "How much would you like to add as Balance?" <br />
+your input: `BALANCE` Note: Balance has to be a valid number. <br />
+
+
+Examples:
+
+```agsl
+add-account
+>>What is your name?
+Steve
+>>How much would you like to add as Balance?
+1000
+```
+
+
+### List all transactions: `view-transactions-all`
+
+Shows a list of all the transactions processed by the **current** account.
+
+Format: `view-transactions-all`
+
+Example:
+```agsl
+>>view-transactions-all
+Account Name: jenson Transaction Type: deposit Amount: 100 Date: 26/03/2023
+Account Name: jenson Transaction Type: withdraw Amount: 10 Date: 26/03/2023
+```
+
+
+### Depositing amount : `deposit`
+
+Deposits AMOUNT into the users account.
+
+Format: `deposit AMOUNT`
+
+Examples:
+`deposit 100`
+
+### switch current account: `switch-to`
+
+Switches current account to the new requested account, if it exists.
 
-**Q**: How do I transfer my data to another computer? 
+Format: `switch-to NAME`
+
+* The search is case-insensitive. e.g `jane` will match `Jane`
+* Only the name is searched.
+* Only full words will be matched e.g. `Ben` will not match `Benjamin`
+
+Examples:
+```
+switch-to jenson
+```
+
+
+### Add a savings Goal to the current account: `save`
+
+Adds a savings goal to teh current account. <br />
+will raise an alert when attempting to withdraw more than the savings goal, if attempt is before deadline
+
+Format: `save AMOUNT` >> will then prompt the user for the deadline.
+
+* The amount to save has to be a number, -ve numbers alloed -- can be used to overwrite as no savings goal
+* deadline entered has to be in dd-mm-YYYY format. No other format will be entertained by the program
+* any error will only be displayed after the deadline is received and processed.
+
+Examples:
+```
+save 300
+>>What would be the end date for the duration of your Saving goal?
+11-11-2011
+>>try saving a Minimum of $300 until 11-11-2011
```

Comment:
Wait this is really what is printed in the program? Cos try should be Try and Minimum shld be minimum